src/components/tl/mlx5/tl_mlx5_context.c

```diff
@@ -60,7 +71,7 @@ UCC_CLASS_CLEANUP_FUNC(ucc_tl_mlx5_context_t)
     }
 
     if (ucc_tl_mlx5_remove_shared_ctx_pd(self) != UCC_OK) {
-        tl_error(self->super.super.lib, "failed to free ib ctx and pd");
```

Comment:
this if section does nothing other than printing debug. Why not move this debug print to the ucc_tl_mlx5_remove_shared_ctx_pd itself?